internal/controller/vrg_volsync.go

```diff
@@ -142,13 +142,13 @@ func (v *VRGInstance) reconcilePVCAsVolSyncPrimary(pvc corev1.PersistentVolumeCl
 
 	protectedPVC := v.findProtectedPVC(pvc.Namespace, pvc.Name)
 	if protectedPVC == nil {
-		protectedPVC = newProtectedPVC
-		v.instance.Status.ProtectedPVCs = append(v.instance.Status.ProtectedPVCs, *protectedPVC)
+		protectedPVC = v.addProtectedPVC(newProtectedPVC.Namespace, newProtectedPVC.Name)
```

Comment:
Is newProtectedPVC new empty object? or object with some fields set? Previously we copied the object into the instance here. Now we add a new empty object.

Reply:
The newProtectedPVC is created before:
	newProtectedPVC := &ramendrv1alpha1.ProtectedPVC{
		Name:               pvc.Name,
		Namespace:          pvc.Namespace,
		ProtectedByVolSync: true,
		StorageClassName:   pvc.Spec.StorageClassName,
		Annotations:        protectedPVCAnnotations(pvc),
		Labels:             pvc.Labels,
		AccessModes:        pvc.Spec.AccessModes,
		Resources:          pvc.Spec.Resources,
	}
So, currently the logic is:
- try to find a protectedPVC, if not found - create a new protectedPVC 
- if protectedPVC found, but differs from newProtectedPVC - copy only conditions from protectedPVC to newProtectedPVC
- in both cases invoke DeepCopyInto from newProtectedPVC to protectedPVC
Seems, that it can be simplified more

Reply:
Strange that we copy the conditions from the existing protected PVC, but otherwise copy everything else from the new protected pvc.

I think the code should do this instead:

1. Get the existing protected pvc
2. If missing, add a new empty one
3. Update the fields of the protected pvc

Reply:
@BenamarMk , can you shed some light on this code fragment logic mystery, please?

Reply:
Do you want me to explain the old code or the new one? I don't see a bug in the old code. Of course you can try to simplify the blocks.

Reply:
That block had an ephemeral bug, but the next reconciliation fixes it.Thank you. 
Simplifying that function is a good idea.